src/components/DropdownBase/DropdownBase.js

```diff
@@ -31,6 +31,13 @@ export default class DropdownBase extends React.Component {
     if (this.props.syncTextWithSelectedItem) this.setState({ text: selectedItem });
   }
 
+  componentDidUpdate(prevProps) {
+    if (prevProps.text !== this.props.text) {
+      // eslint-disable-next-line react/no-did-update-set-state
```

Comment:
We might need to come up with a different approach to avoid setting state in `componentDidUpdate` hook.
Can we avoid using `this.state.text` on line 71 altogether.
Why don't we use `this.props.text` directly and avoid state change?

Reply:
I haven't tried this approach. I'll take a look into it.